packages/tools/devtools/devtools-view/src/components/data-visualization/EditableView.tsx

```diff
@@ -0,0 +1,364 @@
+				onOptionSelect={onOptionSelect}
+				value={
+					editProp === undefined
+						? node.value === null
+							? "null"
+							: typeof node.value
+						: editProp.type
+				}
+				selectedOptions={[
+					editProp === undefined
+						? node.value === null
+							? "null"
+							: typeof node.value
+						: editProp.type,
+				]}
+			>
+				{options.map((option) => (
+					<Option
+						onMouseDownCapture={(event): void => {
+							event.preventDefault();
+						}}
+						key={option}
+					>
+						{option}
+					</Option>
+				))}
+			</Dropdown>
+			{innerView}
+		</div>
+	);
+}
+
+/**
+ * Component which allows for boolean editing
+ */
+function EditableBooleanComponent(props: EditableComponentProps): React.ReactElement {
+	const { node, submitChange } = props;
+
+	/**
+	 * Auxiliary function to make boolean editing buttons
+	 * @param isTrue - Determines if it should be a true or false button
+	 * @returns 2 buttons, one for setting value to false and one for true
+	 */
+	function boolButton(isTrue: boolean): React.ReactElement {
+		return (
+			<Button
+				size="small"
+				disabled={isTrue === node.value}
+				onClick={(event): void => {
+					event.preventDefault();
+					event.stopPropagation();
+					submitChange(isTrue);
+				}}
+			>
+				{String(isTrue).charAt(0).toUpperCase() + String(isTrue).slice(1)}
+			</Button>
+		);
+	}
+	return (
+		<>
+			{boolButton(true)}
+			{boolButton(false)}
+		</>
+	);
+}
+
+/**
+ * Input for {@link EditableInputComponent}
+ */
+interface EditableInputComponent extends EditableComponentProps {
+	inputType: "string" | "number";
+}
+
+/**
+ * Component which allows for number editing
+ */
+function EditableInputComponent(props: EditableInputComponent): React.ReactElement {
+	const { node, submitChange, inputType, editProp, setEditProp } = props;
+
+	/**
+	 * Provides confirming and canceling an edit functionality
+	 */
+	const onKeyDown = React.useCallback(
+		(event: React.KeyboardEvent<HTMLInputElement>) => {
+			const key = event.key;
+
+			if (editProp === undefined) {
+				return;
+			}
+			if (key === "Enter") {
+				submitChange(editProp.type === "number" ? Number(editProp.value) : editProp.value);
+				event.currentTarget.blur();
+			}
+
+			if (key === "Escape") {
+				event.currentTarget.blur();
+			}
+		},
+		[editProp, submitChange],
+	);
+
+	/**
+	 * Sets the local data to what the user wrote in the component
+	 */
+	function onChange(_ev: React.ChangeEvent<HTMLInputElement>, data: InputOnChangeData): void {
+		setEditProp({
+			type: editProp === undefined ? typeof node.value : editProp.type,
+			value: data.value,
+		});
+	}
+
+	return (
+		<Input
+			size="small"
+			appearance="underline"
+			contentEditable={true}
+			value={editProp !== undefined ? (editProp.value as string) : (node.value as string)}
+			onKeyDown={onKeyDown}
+			onChange={onChange}
+			type={inputType === "string" ? "text" : "number"}
+		/>
+	);
+}
+
+const allEdits = [
```

Comment:
Nit: can we add a brief comment to this?

Reply:
Further nitpick, but just a nitpick: I would recommend strictly documenting this in terms of what it means. E.g. something like: "The list of all edit types supported by the view". That way if we end up using this in other places later, we don't have to update the docs.